airflow/providers/apache/beam/operators/beam.py

```diff
@@ -180,7 +180,7 @@ def __init__(
         py_system_site_packages: bool = False,
         gcp_conn_id: str = "google_cloud_default",
         delegate_to: Optional[str] = None,
-        dataflow_config: Optional[Union[DataflowConfiguration, dict]] = None,
+        dataflow_config: Union[DataflowConfiguration, dict] = {},
```

Comment:
[Don’t do this](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments); keep using the optional form and do

```python
if dataflow_config is None:
    self.dataflow_config = DataflowConfiguration()
elif isinstance(dataflow_config, dict):
    self.dataflow_config = DataflowConfiguration(**dataflow_config)
else:
    self.dataflow_config = dataflow_config
```

Reply:
Fixed